crates/wasmtime/src/func.rs

```diff
@@ -1456,15 +1456,11 @@ pub struct Caller<'a> {
 impl Caller<'_> {
     /// Looks up an export from the caller's module by the `name` given.
     ///
-    /// Note that this function is only implemented for the `Extern::Memory`
-    /// type currently. No other exported structure can be acquired through this
-    /// just yet, but this may be implemented in the future!
```

Comment:
I think that the documentation on the `Caller` type itself needs to be updated to indicate it's more than just memory now, and this function can probably point to that documentation to guide about usages of the function.

On the `Caller` struct itself I think it can be updated to something like:

> This structure can be taken as the first parameter of a closure passed to
> [`Func::wrap`], and it can be used to learn information about the caller of
> the function, such as the calling module's memory, exports, etc.
>
> The primary purpose of this structure is to provide access to the
> caller's information, namely it's exported memory and exported functions. This
> allows functions which take pointers as arguments to easily read the memory the
> pointers point into, or if a function is expected to call `malloc` in the wasm
> module to reserve space for the output you can do that.
>
> Note that this `Caller` type a pretty temporary mechanism for accessing the
> caller's information until interface types has been fully standardized and
> implemented. The interface types proposal will obsolete this type and this will
> be removed in the future at some point after interface types is implemented. If
> you're relying on this `Caller` type it's recommended to become familiar with
> interface types to ensure that your use case is covered by the proposal.

Reply:
Great, will make these updates. Thank you!